<a href="https://colab.research.google.com/github/litalentz/Amazon-Seller-Dataset-Analysis/blob/main/CSI-260_Adv_Python_Pandas_Data_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Final Project: Pandas Data Analysis for Ann Clark Amazon Data Sheet

Author: Amelita Lentz
Class: CSI-260-02
Assignment: Final Project
Due Date: Dec. 12, 2022 11:59 PM

Certification of Authenticity:
I certify that this is entirely my own work, except where I have given
fully-documented references to the work of others. I understand the definition
and consequences of plagiarism and acknowledge that the assessor of this
assignment may, for the purpose of assessing this assignment:
- Reproduce this assignment and provide a copy to another member of academic
- staff; and/or Communicate a copy of this assignment to a plagiarism checking
- service (which may then retain a copy of this assignment on its database for
- the purpose of future plagiarism checking)

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

from google.colab import drive
drive.mount('/content/drive')
data = pd.read_excel(r'/content/drive/MyDrive/CSI-260 Advanced Python/BusinessReport.xlsx')


I am using a dataset encompassing Amazon seller data over a year long period from the company Ann Clark's Cookie Cutters.

Lets give you an idea of what this data includes.

First, what the data file looks like, specifically the titles and what type of data they hold, and any null values.

In [ ]:
data.head()

Title  SessionsTotal  \
0  Professional-Grade Food Coloring Gel Made in U...         236799   
1  Happy Gingerbread Man Cookie Cutter, Large 5.25"           33158   
2                    Easter Bunny Cookie Cutter, 4"           32081   
3  St. Patrick's Day Irish Shamrock Cookie Cutter...          23566   
4  9-Piece Numbers Cookie Cutter Set Birthday Num...          51167   

   SessionsTotalB2B  SessionPercentageTotal  SessionPercentageTotalB2B  \
0              2037                  0.0533                     0.1331   
1               108                  0.0075                     0.0071   
2                 1                  0.0072                     0.0001   
3                 9                  0.0053                     0.0006   
4               160                  0.0115                     0.0105   

   PageViewsTotal  PageViewsTotalB2B  PageViewsPercentageTotal  \
0          340696               2778                    0.0542   
1           62598                127                    0.0100   
2           43576                  1                    0.0069   
3           31654                 19                    0.0050   
4           74440                247                    0.0118   

   PageViewsPercentageTotalB2B  FeaturedOfferBuyBoxPercentage  \
0                       0.1398                         0.9861   
1                       0.0064                         0.7286   
2                       0.0001                         0.9910   
3                       0.0010                         0.9873   
4                       0.0124                         0.9641   

   FeaturedOfferBuyBoxPercentageB2B  UnitsOrdered  UnitsOrderedB2B  \
0                            0.9942         35567              972   
1                            0.9685         10015              201   
2                            0.0000          9061              138   
3                            0.8947          8985              192   
4                            0.9636          8342              151   

   UnitSessionPercentage  UnitSessionPercentageB2B  OrderedProductSales  \
0                 0.1502                         0            792483.74   
1                 0.3020                         0             69707.15   
2                 0.2824                         0             61811.51   
3                 0.3813                         0             62509.58   
4                 0.1630                         0            129755.61   

   OrderedProductSalesB2B  TotalOrderItems  TotalOrderItemsB2B  
0                21958.71            35003                 868  
1                 1238.99             9332                 161  
2                  807.86             8783                 126  
3                 1130.50             8726                 158  
4                 2261.34             8305                 147

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 780 entries, 0 to 779
Data columns (total 19 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Title                             780 non-null    object 
 1   SessionsTotal                     780 non-null    int64  
 2   SessionsTotalB2B                  780 non-null    int64  
 3   SessionPercentageTotal            780 non-null    float64
 4   SessionPercentageTotalB2B         780 non-null    float64
 5   PageViewsTotal                    780 non-null    int64  
 6   PageViewsTotalB2B                 780 non-null    int64  
 7   PageViewsPercentageTotal          780 non-null    float64
 8   PageViewsPercentageTotalB2B       780 non-null    float64
 9   FeaturedOfferBuyBoxPercentage     780 non-null    float64
 10  FeaturedOfferBuyBoxPercentageB2B  780 non-null    float64
 11  UnitsOrdered                      780 non-null    int64  
 12  UnitsOrd

Note: The column titles are all in camel case to make coding with them easier.

○ Title refers to the name of the product, ex. Pumpkin Cookie Cutter

○ Units Ordered is the number of units purchased whereas Total Order Items is the number of orders places.

○ B2B means business to business

○ The Featured Offer (Buy Box) is a coveted Amazon spot that essentially helps sell your product. The percentage is calculated as the total page views in which you are the Featured Offer divided by the total page views received by products you list.


In [ ]:
print(data.isnull().sum())

Title                               0
SessionsTotal                       0
SessionsTotalB2B                    0
SessionPercentageTotal              0
SessionPercentageTotalB2B           0
PageViewsTotal                      0
PageViewsTotalB2B                   0
PageViewsPercentageTotal            0
PageViewsPercentageTotalB2B         0
FeaturedOfferBuyBoxPercentage       0
FeaturedOfferBuyBoxPercentageB2B    0
UnitsOrdered                        0
UnitsOrderedB2B                     0
UnitSessionPercentage               0
UnitSessionPercentageB2B            0
OrderedProductSales                 0
OrderedProductSalesB2B              0
TotalOrderItems                     0
TotalOrderItemsB2B                  0
dtype: int64


There are none! The dataset was decently complete when I received it, I did a little bit of further touch-ups in excel.

Now that you have a basic knowledge of what kind of data we're looking at, lets delve into the summary statistics.

In [ ]:
data.describe()

SessionsTotal  SessionsTotalB2B  SessionPercentageTotal  \
count     780.000000        780.000000              780.000000   
mean     5688.756410         19.612821                0.001281   
std     10622.406687         79.044089                0.002392   
min         3.000000          0.000000                0.000000   
25%      1323.750000          2.000000                0.000300   
50%      3186.500000          7.000000                0.000700   
75%      6772.500000         18.000000                0.001500   
max    236799.000000       2037.000000                0.053300   

       SessionPercentageTotalB2B  PageViewsTotal  PageViewsTotalB2B  \
count                 780.000000      780.000000         780.000000   
mean                    0.001286     8057.538462          25.470513   
std                     0.005165    15535.346083         107.130685   
min                     0.000000        5.000000           0.000000   
25%                     0.000100     1698.250000           2.000000   
50%                     0.000500     4280.500000          10.000000   
75%                     0.001200     9255.000000          23.000000   
max                     0.133100   340696.000000        2778.000000   

       PageViewsPercentageTotal  PageViewsPercentageTotalB2B  \
count                780.000000                   780.000000   
mean                   0.001282                     0.001297   
std                    0.002471                     0.005389   
min                    0.000000                     0.000000   
25%                    0.000300                     0.000100   
50%                    0.000700                     0.000500   
75%                    0.001500                     0.001200   
max                    0.054200                     0.139800   

       FeaturedOfferBuyBoxPercentage  FeaturedOfferBuyBoxPercentageB2B  \
count                     780.000000                        780.000000   
mean                        0.961820                          0.799426   
std                         0.059086                          0.375789   
min                         0.647300                          0.000000   
25%                         0.962275                          0.909100   
50%                         0.987600                          1.000000   
75%                         0.993400                          1.000000   
max                         1.000000                          1.000000   

       UnitsOrdered  UnitsOrderedB2B  UnitSessionPercentage  \
count    780.000000       780.000000             780.000000   
mean    1109.661538        23.756410               0.191905   
std     1777.189828        44.261385               0.074326   
min        1.000000         0.000000               0.000100   
25%      230.000000         4.000000               0.139350   
50%      643.500000        14.000000               0.196250   
75%     1406.750000        31.000000               0.246300   
max    35567.000000       972.000000               0.381300   

       UnitSessionPercentageB2B  OrderedProductSales  OrderedProductSalesB2B  \
count                     780.0           780.000000              780.000000   
mean                        0.0         10720.546282              222.084295   
std                         0.0         30776.337555              829.130622   
min                         0.0             6.990000                0.000000   
25%                         0.0          1885.735000               29.980000   
50%                         0.0          5148.715000              107.305000   
75%                         0.0         11933.255000              239.007500   
max                         0.0        792483.740000            21958.710000   

       TotalOrderItems  TotalOrderItemsB2B  
count       780.000000          780.000000  
mean       1081.102564           20.480769  
std        1737.580155           37.964187  
min           1.000000            0.000000  
25%         22

Note: Computations done with any percentage column will spit back ex. the numerical .05 instead of the percentage 5%.

You can see in the summary statistics the distribution of data present in the columns. For example, Sessions Total is positively skewed to the right, meaning that the mean is greater than the median. Also notice how that the maximun value is around 231,111 greater than the mean. This points to outliers in Sessions Total. Scanning over the rest of the data, there are some cases in which the mean is even double the median.

To visualize this better lets create a histogram for Sessions Total.

The histogram below shows the skewed distribution with a max around 236K and (excluding a few more outliers) the rest of the values to be less than 26K. Around half of all of the data values sit under 3k.

In [ ]:
fig = px.histogram(data, x='SessionsTotal')

fig.show()

Lets take care of all of the outliers in the dataset together!

Notice the data does not follow a normal distribution. Since the data is skewed, instead of using a z-score we can use interquartile range (IQR) to determine the outliers.

In [ ]:
q1 = data.quantile(0.25)

q3 = data.quantile(0.75)

IQR = q3 - q1
not_outliers = data[~((data<(q1-1.5*IQR)) | (data>(q3+1.5*IQR)))]
data_no = not_outliers.dropna().reset_index()


<ipython-input-110-7e70e1c70259>:6: FutureWarning:

Automatic reindexing on DataFrame vs Series comparisons is deprecated and will raise ValueError in a future version.  Do `left, right = left.align(right, axis=1, copy=False)` before e.g. `left == right`



I decided to drop the outliers instead of changing the values or treating them as null. There are reasons to why those specific products are outliers. Exluding them completely allows us to focus on the more relevent data without influencing possible finding by including 'false' generated values.

With the outliers taken care of, we can delve into specific areas of inquiry.

First lets take a look a the relationship between Total Ordered Items and Total Page Views.
To show the impact of excluding the outliers, I've included what the scatterplot would have looked like prior to our culling.

In [ ]:
fig_old = px.scatter(data_frame = data, x="TotalOrderItems",
                    y="PageViewsTotal",
                    trendline="ols",
                    title="Relationship between Total Ordered Items & Total Page Views")
fig_old.show()

fig_pv_oi = px.scatter(data_frame = data_no, x="TotalOrderItems",
                    y="PageViewsTotal",
                    trendline="ols",
                    title="Relationship between Total Ordered Items & Total Page Views")
fig_pv_oi.show()

Here we can see that there seems to be a strong positive linear relationship between the Page Views and Ordered Items as all the points follow the trend line closely.

Lets calculate the correlation coefficent ...

In [ ]:
# If we wanted to see all the correlation coefficents in a table, follow this code below
correlation_table = data_no.corr()

# For our purposes right now we only need to look at the Page Views and Ordered Items so ...
correlation_table['PageViewsTotal'].sort_values(ascending=False)

PageViewsTotal                      1.000000
PageViewsPercentageTotal            0.998906
SessionsTotal                       0.997413
SessionPercentageTotal              0.996187
OrderedProductSales                 0.940271
TotalOrderItems                     0.892834
UnitsOrdered                        0.890339
OrderedProductSalesB2B              0.830764
TotalOrderItemsB2B                  0.797119
UnitsOrderedB2B                     0.757639
SessionsTotalB2B                    0.641198
SessionPercentageTotalB2B           0.640314
PageViewsPercentageTotalB2B         0.631348
PageViewsTotalB2B                   0.630937
UnitSessionPercentage               0.096640
FeaturedOfferBuyBoxPercentageB2B   -0.138181
FeaturedOfferBuyBoxPercentage      -0.171835
index                              -0.861212
UnitSessionPercentageB2B                 NaN
Name: PageViewsTotal, dtype: float64

Here we can see that the correlation is strong with a value of 0.89

All of this means that products with high Page Views are likely to be the highest selling products and vice versa. This data can point to where the company needs to improve for increased conversions, i.e. boosting promotions on it's lower viewed items in an attempt to increase their sales or cutting them entirely depending on how poorly the sales are. On the other hand, Ann Clark could promote its higher selling goods to increase Page Views and, therefore, its Ordered Items.

Note: Correlation ≠ Causation

Looking more at these correlation coefficents we can draw some more conclusions ...

○ Page Views has a stellar correlation with Sessions Total, to be expected as they are similar metrics.

○ Sales, Units Ordered, and Ordered Items are all very similar in their correlations to Page Views, again to be expected as they are similar metrics. Sales is the highest at .94 meaning that the sales revenue is tightly tied to the amount of page views a product has received.

○ The correlation starts to fall from excellent to good as we look at the B2B variables.

In [ ]:
fig = px.scatter(data_frame = data_no, x="SessionsTotal",
                    y="PageViewsTotal",
                    trendline="ols",
                    title="Relationship between Sessions Total & Total Page Views")
fig.show()

In [ ]:
fig_pv = px.scatter(data_frame = data_no, x="PageViewsTotal",
                    y="OrderedProductSales",
                    trendline="ols",
                    title="Relationship between Total Page Views & Ordered Product Sales")
fig_pv.show()

fig_ss = px.scatter(data_frame = data_no, x="SessionsTotal",
                    y="OrderedProductSales",
                    trendline="ols",
                    title="Relationship between Sessions Total & Ordered Product Sales")
fig_ss.show()

The relationship between Sessions and Sales is practically identical to that of Page Views and Sales. This futhers the point that the two metrics are intertwined and Ann Clark should be considering both, session for topline and pageviews for conversion.

In [ ]:
price_per = data_no['OrderedProductSales'] / data_no['UnitsOrdered']

fig_pp = px.scatter(data_frame = data_no, x=price_per,
                    y='UnitsOrdered',
                    title="Relationship between the Price & Units Ordered")
fig_pp.show()


Looking at the Price per Unit compared to the Units Ordered there doesn't seem to be a strong linear realationship. This could be due to the fact that some products are the same price, however some are more popular than others.

In [ ]:
data_no['Price'] = price_per
sorted_price = data_no.sort_values(by='Price', ascending=False) # greatest to least
print(sorted_price['Title'].head())
print(sorted_price['Price'].head())
print(sorted_price['UnitsOrdered'].head())

234    Perfect Cookie Rolling Pin 1/4-in. Fixed Depth...
431    Mardi Gras Cookie Decorating Kit, 8-pc. Basic ...
434    St. Patrick's Day Cookie Decorating Kit Cookie...
409                     Valentine’s Day Sprinkles 5 Pack
433    Easter Cookie Decorating Kit Cookie Cutters, F...
Name: Title, dtype: object
234    40.183709
431    27.381304
434    25.990000
409    19.990000
433    19.305789
Name: Price, dtype: float64
234    585.0
431     23.0
434     15.0
409     94.0
433     19.0
Name: UnitsOrdered, dtype: float64


In [ ]:
sorted_sold = data_no.sort_values(by='UnitsOrdered', ascending=False) # greatest to least
print(sorted_sold['Title'].head())
print(sorted_sold['Price'].head())
print(sorted_sold['UnitsOrdered'].head())

0                Cute Elephant Cookie Cutter, 4.25"
1    Easter Bunny Cookie Cutter, 4.25" Made in USA 
4          Goldfish / Simple Fish Cookie Cutter, 3"
3                Snowman with Hat Cookie Cutter, 4"
2                      Sea Turtle Cookie Cutter, 4"
Name: Title, dtype: object
0    7.106249
1    6.782416
4    7.126784
3    6.904435
2    7.106709
Name: Price, dtype: float64
0    3154.0
1    2873.0
4    2861.0
3    2850.0
2    2847.0
Name: UnitsOrdered, dtype: float64


Above you can compare the Price to the Units Ordered. The first section showcases the dataset variables Title, Price, and Units Ordered ordered by Price. The second looks at the same dataset varaibles now ordered by the number of Units Ordered.

Note:

○ None of the top products are the same.

○ The top unit sellers sell at a price between 6.78 - 7.11 dollars and with units being ordered from 2847 - 3154

○ The top priced products sell at a price between 19.31 - 40.18 dollars and with units being ordered from 15 - 585

In [ ]:
data_no['Price'].describe()

count    445.000000
mean       8.835955
std        3.310676
min        6.077771
25%        7.034246
50%        7.126103
75%       10.647786
max       40.183709
Name: Price, dtype: float64

The price description above gives you some context.